src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -211,51 +222,104 @@ private static int identifier(String s, int startFrom) {
       OutputSpec.LoopStatement loop,
       MethodContext context,
       SampleValueSet valueSet,
-      Map<String, TypeModel> localVars) {
+      ScopeTable localVars) {
 
+    ScopeTable scope = localVars.newChild();
     OutputView.VariableView accessor =
-        accessor(loop.getCollection(), context, valueSet, localVars, loop.getVariable());
+        accessorNewVariable(loop.getCollection(), context, valueSet, scope, loop.getVariable());
     OutputView.LoopView ret =
         OutputView.LoopView.newBuilder()
             .variableType(
                 context
                     .getNamer()
-                    .getAndSaveTypeName(context.getTypeTable(), localVars.get(loop.getVariable())))
+                    .getAndSaveTypeName(context.getTypeTable(), scope.get(loop.getVariable())))
             .variableName(context.getNamer().localVarName(Name.from(loop.getVariable())))
             .collection(accessor)
             .body(
                 loop.getBodyList()
                     .stream()
-                    .map(body -> toView(body, context, valueSet, localVars))
+                    .map(body -> toView(body, context, valueSet, scope))
                     .collect(ImmutableList.toImmutableList()))
             .build();
 
     // The variable is only visible within the loop, delete it from the table before we return.
-    localVars.remove(loop.getVariable());
     return ret;
   }
 
-  private static OutputView.AssignmentView assignmentView(
-      String assign,
-      MethodContext context,
-      SampleValueSet valueSet,
-      Map<String, TypeModel> localVars) {
-
-    int p = identifier(assign, 0);
-    String ident = assign.substring(0, p);
+  private static OutputView.DefinitionView definitionView(
+      String definition, MethodContext context, SampleValueSet valueSet, ScopeTable localVars) {
+    int p = identifierEnd(definition, 0);
+    String ident = definition.substring(0, p);
```

Comment:
what do you think about invoking `trim` here and in line 260 (`definition.substring(p+1)`, though this will need to be tweaked if we do) so that we allow whitespace around the `=`?

(Even if you agree it's worthwhile, I'm OK deferring)

Reply:
If we're doing this, I think we should also allow spaces like `foo = $resp .bar [ 42   ]`. We can definitely do that with a lexer, but I think we should defer.